tests/java/lang/path_traversal/testdata/main.java

```diff
@@ -0,0 +1,44 @@
+// Use bearer:expected java_lang_path_traversal to flag expected findings
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class Foo {
+  public void bad(HttpServletRequest request, HttpServletResponse response) {
+    String safeDirectory = "/safe_dir/";
+    // bearer:expected java_lang_path_traversal
+    Path fullPath = Paths.get(request.getParameter("filepath"));
+
+    if (fullPath.startsWith(Paths.get(safeDirectory))) {
+      // ...
+    }
+  }
+
+  public void bad2(HttpServletRequest request, HttpServletResponse response) {
+    String safeDirectory = "/safe_dir/";
+    String filename = request.getParameter("filepath");
+    // bearer:expected java_lang_path_traversal
+    Path fullPath = Paths.get("tmp/user/" + filename);
+
+    if (fullPath.startsWith(Paths.get(safeDirectory))) {
+      // ...
+    }
+  }
+
+  public void ok(HttpServletRequest request, HttpServletResponse response) {
+    String safeDirectory = "/safe_dir/";
+    String filepath = "/tmp/user/bar.txt";
+    // Path fullPath = Paths.get(request.getParameter("filepath")).normalize();
```

Comment:
The rule is broad - if we uncomment this, `Path.get(USER_INPUT).normalize() ` is present and no findings are reported. 

Is there a way to limit this @cfabianski ?

Reply:
^ We needed to use a sanitizer here, instead of a "match on absence" trigger.